apps/web/pages/event-types/[type]/index.tsx

```diff
@@ -99,33 +92,38 @@ const querySchema = z.object({
 });
 
 export type EventTypeSetupInfered = inferSSRProps<typeof getServerSideProps>;
+export type EventTypeSetupProps = RouterOutputs["viewer"]["eventTypes"]["get"];
 
 const EventTypePage = (props: inferSSRProps<typeof getServerSideProps>) => {
   const { t } = useLocale();
+  const utils = trpc.useContext();
+  const router = useRouter();
+  const { tabName } = querySchema.parse(router.query);
+
   const { data: eventTypeApps } = trpc.viewer.apps.useQuery({
     extendsFeature: "EventType",
   });
 
-  const { eventType: dbEventType, locationOptions, team, teamMembers } = props;
-  // TODO: It isn't a good idea to maintain state using setEventType. If we want to connect the SSR'd data to tRPC, we should useQuery(["viewer.eventTypes.get"]) with initialData
-  // Due to this change, when Form is saved, there is no way to propagate that info to eventType (e.g. disabling stripe app doesn't allow recurring tab to be enabled without refresh).
-  const [eventType, setEventType] = useState(dbEventType);
-
-  const router = useRouter();
-  const { tabName } = querySchema.parse(router.query);
+  const { data, isLoading } = trpc.viewer.eventTypes.get.useQuery({ id: props.type });
```

Comment:
It would be better to move useQuery to a wrapper Component for `EventTypePage` component. So, loading logic would be limited to the wrapper component and thus EventTypePage would have non null eventType. 
This would avoid all the newly added `eventType?` and `eventType&&` things

Reply:
We should use initialData with useQuery that is obtained from SSR. That makes the first page load really fast with no loader and later navigations to eventType page would be client side rendered.

Reply:
Done